tasks/storage_deal.go

```diff
@@ -156,6 +160,10 @@ func MakeStorageDeal(ctx context.Context, config NodeConfig, node api.FullNode,
 }
 
 func logStages(info api.DealInfo, log UpdateStatus) {
+	if info.DealStages == nil {
+		log("Deal stages is nil")
```

Comment:
This would still panic, no? Needs a return of some sort.

Reply:
🤦 of course.

We could probably remove this after a while, but there are still a lot of clients/miners running older versions where `DealStages` is nil.